### PR TITLE
Add FXIOS-6664 [v117] Handle ETP coordinator is deinit

### PR DESCRIPTION
**Pull request by aalaaeid**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6664)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/14892)

## :bulb: Description

1. this handling decollation ETP coordinator upon dismissal ETPMenu Cases
2. Turning ETP coordinator refactor ON in developer builds.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift [hunk 1]
```diff
@@ -42,6 +42,10 @@ class EnhancedTrackingProtectionCoordinator: BaseCoordinator,
         enhancedTrackingProtectionMenuVC.enhancedTrackingProtectionMenuDelegate = self
     }
 
+    deinit {
+        print("ETPCoordinator is deinit")
+    }
```
Comment (razvanlitianu) [inline Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:47]: Was this print for testing purposes?

File: nimbus-features/etpCoordinatorRefactor.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ features:
           enabled: false
       - channel: developer
         value: 
-          enabled: false
+          enabled: true
```
Comment (lmarceau) [inline nimbus-features/etpCoordinatorRefactor.yaml:18]: So discussing with Orla, turns out we could turn this ON for all channel 🎉 So true by default, and true on beta!

**Review by lmarceau — COMMENTED** [review]
One small comment, on top of that we should remove the print, and all looks good to me!

File: Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift (hunk 1 above)
Comment (aalaaeid) [reply Client/Coordinators/EnhancedTrackingProtectionCoordinator.swift:47]: Yeah, will remove that

**Review by yoanarios — APPROVED** [review]

**Pull request merged**